GDevelop.js/Bindings/Bindings.idl

```diff
@@ -166,25 +171,44 @@ interface PairStringVariable {
     [Ref] Variable WRAPPED_GetVariable();
 };
 
+enum Variable_TYPES {
+    "Variable_TYPES::STRING",
+    "Variable_TYPES::NUMBER",
+    "Variable_TYPES::BOOLEAN",
+    "Variable_TYPES::STRUCTURE",
+    "Variable_TYPES::ARRAY"
+};
+
+interface VariablePointer {
+    Variable get();
+};
+
 interface Variable {
     void Variable();
 
+    Variable_TYPES GetType();
+
     void SetString([Const] DOMString str);
     [Const, Ref] DOMString GetString();
     void SetValue([Const] double val);
     double GetValue();
+    void SetBool([Const] boolean val);
+    boolean GetBool();
 
     boolean HasChild([Const] DOMString str);
     [Ref] Variable GetChild([Const] DOMString str);
     void RemoveChild([Const] DOMString name);
     boolean RenameChild([Const] DOMString oldName, [Const] DOMString newName);
     [Value] VectorString GetAllChildrenNames();
     unsigned long GetChildrenCount();
-    boolean IsNumber();
-    boolean IsStructure();
     boolean Contains([Const, Ref] Variable variableToSearch, boolean recursive);
     void RemoveRecursively([Const, Ref] Variable variableToRemove);
 
+    boolean HasIndex([Const] unsigned long index);
+    [Ref] Variable GetIndex([Const] unsigned long index);
+    void RemoveIndex([Const] unsigned long index);
+    [Const, Ref] VectorVariable GetAllChildrenList();
```

Comment:
Let's see if we can avoid exposing this. This is exposing the internal of the class, and it's dangerous/painful to use from JS. Might as well avoid it entirely :)

Reply:
I thought those might be needed for the initial variables editors?

Reply:
If it's used then no problem, if it's not used let's remove it :)

Reply:
Especially because `VectorVariable` is actually hiding a vector to smart pointers, which is freaking me out - it's so easy to forget a ref, get a copy, never release it, get a memory leak.